src/event.rs

```diff
@@ -3,6 +3,25 @@
 //! These are sent to the closure given to [`EventLoop::run(...)`][event_loop_run], where they get
 //! processed and used to modify the program state. For more details, see the root-level documentation.
 //!
+//! Some of these events represent different "parts" of a traditional event-handling loop.  You could
+//! approximate the basic ordering loop of [`EventLoop::run(...)`] like this:
+//!
+//! ```rust,no_run
+//! let control_flow = &mut ControlFlow::Wait;
+//! event_handler(NewEvents(StartCause::Init), ..., control_flow);
+//! while *control_flow != ControlFlow::Exit {
+//!     event_handler(NewEvents(StartCause::Something), ..., control_flow);
+//!     for e in (window events, user events, device events) {
+//!         event_handler(e, ..., control_flow);
+//!     }
+//!     event_handler(EventsCleared, ..., control_flow);
```

Comment:
Yeah I didn't try to make a *complete* example, more a cartoon of how the pieces fit together... because I didn't understand the complete picture.  Do we want a full and more complex example?

Reply:
I'd like to wait for #1041 to get resolved, since a full example that includes the current redraw complexity is complex enough that it would largely defeat the point of having this sort of example.

Reply:
Since it looks like #1041 might take a fair bit of work, should I remove those bits of the docs so we can merge this, or should I update them to reflect the current model so we can merge this?

Reply:
Could you remove this for now, but open another PR adding the example once the #1041 changes have been merged?

Reply:
Coming back to this - we've got a `redraw-requested-2.0` branch that implements the new API. There's a clear model for the redraw API there, so if you'd like to re-target this PR against that branch we should be able to merge it promptly.

Reply:
Life is rather busy now, so this is squarely in the "when I can get around to it" category.

Reply:
The `redraw-requested-2.0` branch has been merged onto master.